<launch>

  <arg name="drone_id" default="0"/>
  <arg name="drone_num" default="12"/>
  <arg name="commander" default="true"/>

  <node pkg="ros_proxy" name="commander_ros_proxy" type="ros_proxy_safmc" output="screen" >
  
    <remap from="~my_odom" to="/drone_$(arg drone_id)_mavros/local_position/odom"/>
    <!-- Peter Modification -->
    <!-- <remap from="/goal" to="/move_base_simple/goal"/> -->

    <param name="is_commander" value="$(arg commander)"/>
    <param name="commander_ip" value="tcp://127.0.0.1:18099" type="string"/>

    <param name="drone_id" value="$(arg drone_id)"/>
    <param name="drone_num" value="$(arg drone_num)"/>
    <!-- <param name="drone_ip_0" value="tcp://192.168.2.20:18010" type="string"/>
    <param name="drone_ip_1" value="tcp://192.168.2.21:18011" type="string"/>
    <param name="drone_ip_2" value="tcp://192.168.2.22:18012" type="string"/>
    <param name="drone_ip_3" value="tcp://192.168.2.23:18013" type="string"/>
    <param name="drone_ip_4" value="tcp://192.168.2.24:18014" type="string"/>
    <param name="drone_ip_5" value="tcp://192.168.2.25:18015" type="string"/>
    <param name="drone_ip_6" value="tcp://192.168.2.26:18016" type="string"/>
    <param name="drone_ip_7" value="tcp://192.168.2.27:18017" type="string"/>
    <param name="drone_ip_8" value="tcp://192.168.2.28:18018" type="string"/>
    <param name="drone_ip_9" value="tcp://192.168.2.29:18019" type="string"/> -->
    <!-- OWL Mini 2 -->
    <param name="drone_ip_8" value="tcp://192.168.2.20:18020" type="string"/>
    <param name="drone_ip_9" value="tcp://192.168.2.21:18021" type="string"/>
    <param name="drone_ip_10" value="tcp://192.168.2.22:18022" type="string"/>
    <param name="drone_ip_11" value="tcp://192.168.2.23:18023" type="string"/>
    <!-- OWL Mini 3 -->
    <param name="drone_ip_0" value="tcp://192.168.2.30:18030" type="string"/>
    <param name="drone_ip_1" value="tcp://192.168.2.31:18031" type="string"/>
    <param name="drone_ip_2" value="tcp://192.168.2.32:18032" type="string"/>
    <param name="drone_ip_3" value="tcp://192.168.2.33:18033" type="string"/>
    <param name="drone_ip_4" value="tcp://192.168.2.34:18034" type="string"/>
    <param name="drone_ip_5" value="tcp://192.168.2.35:18035" type="string"/>
    <param name="drone_ip_6" value="tcp://192.168.2.36:18036" type="string"/>
    <param name="drone_ip_7" value="tcp://192.168.2.37:18037" type="string"/>
    
    <param name="odom_hz" value="30"/>

  </node>

  <node pkg="tf" type="static_transform_publisher" name="map_world_tf" args="0 0 0 0 0 0.707 0.707 map world 10000" />
  <node pkg="tf" type="static_transform_publisher" name="world_baselink_tf" args="0 0 0 0 0 0 1.0 world base_link 10000" />

   <include file="$(find swarm_rviz)/launch/safmc_rviz.launch">
   </include>

  <!-- <include file="$(find visbot_monitor)/launch/rviz.launch">
  </include> -->


</launch>
